Protoposterior El protoposterior de un escenario es el likelihood del escenario multiplicado por el prior.

G- ¿Cuáles son los protoposteriores de los escenarios 1 y 2?

In [2]:
0.5*2/3
0.5^5*1/3

[1] 0.3333333

[1] 0.01041667

H- ¿Suman 1 los protoposteriores?

> No

Posteriores Las probabilidades posteriores tienen que sumar 1.


I- Para conseguirlo, dividir cada protoposterior por la suma de los protoposteriores. Calcular la probabilidad posterior de los escenarios 1 y 2. Es decir, aplicar la regla de Bayes.

In [4]:
P_H1 <- 2/3
P_H2 <- 1/3

P_obs_dado_H1 <- 0.5
P_obs_dado_H2 <- 0.5^5

P_obs <- P_H1*P_obs_dado_H1 + P_H2*P_obs_dado_H2

Posterior_1 <- P_H1*P_obs_dado_H1 / P_obs
Posterior_2 <- P_H2*P_obs_dado_H2 / P_obs

Posterior_1
Posterior_2

[1] 0.969697

[1] 0.03030303

Percepción El modelo estándar de observador bayesiano para hipótesis discretas asume que la percepción es el escenario con la mayor probabilidad posterior (maximum-a-posteriori o estimación MAP)

J- ¿Sería consistente con el principio Gestaltico de “dirección común”?

> Sí, pues la probabilidad posterior es de 96.97% de que la hipótesis 1 (mismo objeto) sea cierta, muy por encima al de la hipótesis 2 (objetos independientes)

K- Explicar por qué el modelo de observador bayesiano va más allá de la descripción que la psicología de la Gestalt hace del fenómeno.

Notar que en este caso el prior es relativamente poco importante. El peso está en el likelihood. La interpretación de la función de likelihood es en términos de hipótesis: cuando el observador recibe evidencia sensorial, cuál es la probabilidad de obtener esa evidencia cuando el estado del mundo (en este caso, escenarios 1 y 2) toma determinado valor? Cada valor del estado del mundo es una hipótesis, y el likelihood de esa hipótesis cuantifica la creencia que tiene el observador de que las observaciones ocurrieron bajo esa hipótesis. Las probabilidades a priori y posteriores también son distribuciones de creencia, cuyos argumentos son hipótesis.

# Ejercicio 8

En una bolsa hay 4 pelotas. Las pelotas pueden ser rojas o negras pero no se sabe cuántas hay de cada color. Se extraen tres pelotas, de a una y reemplazando en la bolsa después de cada extracción, y se obtiene la secuencia (roja, negra, roja). ¿Cuál es el número más probables de pelotas de cada color? Después se extrae una cuarta y es roja. ¿Cómo cambia el cálculo? Calcular la plausibilidad (probabilidad) de cada posible “hipótesis” sobre el contenido de la bolsa.

In [132]:
# Priors: Antes de sacar ninguna
p_H0 <- 1/5 # 0 rojas 4 negras
p_H1 <- 1/5 # 1 roja  3 negras
p_H2 <- 1/5
p_H3 <- 1/5
p_H4 <- 1/5 # 4 rojas 0 negras

In [133]:
# Saco 1 roja, Hipotesis 0
p_obs1_dado_H0 <- 0
# Hipotesis 1
p_obs1_dado_H1 <- 1/4
# Hipotesis 2
p_obs1_dado_H2 <- 2/4
# Hipotesis 3
p_obs1_dado_H3 <- 3/4
# Hipotesis 4
p_obs1_dado_H4 <- 4/4

p_obs1 <- p_obs1_dado_H0 * p_H0 + p_obs1_dado_H1 * p_H1 + p_obs1_dado_H2 * p_H2 + p_obs1_dado_H3 * p_H3 + p_obs1_dado_H4 * p_H4

In [134]:
# Saco 1 roja, Hipotesis 0
p_H0_dado_obs1 <- p_obs1_dado_H0 * p_H0 / p_obs1
p_H0_dado_obs1

[1] 0

In [135]:
# Hipotesis 1
p_H1_dado_obs1 <- p_obs1_dado_H1 * p_H1 / p_obs1
p_H1_dado_obs1

[1] 0.1

In [136]:
# Hipotesis 2
p_H2_dado_obs1 <- p_obs1_dado_H2 * p_H2 / p_obs1
p_H2_dado_obs1

[1] 0.2

In [137]:
# Hipotesis 3
p_H3_dado_obs1 <- p_obs1_dado_H3 * p_H3 / p_obs1
p_H3_dado_obs1

[1] 0.3

In [138]:
# Hipotesis 4
p_H4_dado_obs1 <- p_obs1_dado_H4 * p_H4 / p_obs1
p_H4_dado_obs1

[1] 0.4

In [139]:
# Actualizo priors luego de primera extracción
p_H0 <- p_H0_dado_obs1 # = 0
p_H1 <- p_H1_dado_obs1
p_H2 <- p_H2_dado_obs1
p_H3 <- p_H3_dado_obs1
p_H4 <- p_H4_dado_obs1

In [140]:
# Saco 1 negra, luego de 1 roja, Hipotesis 0
p_obs2_dado_H0 <- 4/4
# Hipotesis 1
p_obs2_dado_H1 <- 3/4
# Hipotesis 2
p_obs2_dado_H2 <- 2/4
# Hipotesis 3
p_obs2_dado_H3 <- 1/4
# Hipotesis 4
p_obs2_dado_H4 <- 0

p_obs2 <- p_obs2_dado_H0 * p_H0 + p_obs2_dado_H1 * p_H1 + p_obs2_dado_H2 * p_H2 + p_obs2_dado_H3 * p_H3 + p_obs2_dado_H4 * p_H4

In [141]:
# Saco 1 negra, luego de 1 roja, Hipotesis 0
p_H0_dado_obs2 <- p_obs2_dado_H0 * p_H0 / p_obs2
p_H0_dado_obs2

[1] 0

In [142]:
# Hipotesis 1
p_H1_dado_obs2 <- p_obs2_dado_H1 * p_H1 / p_obs2
p_H1_dado_obs2

[1] 0.3

In [143]:
# Hipotesis 2
p_H2_dado_obs2 <- p_obs2_dado_H2 * p_H2 / p_obs2
p_H2_dado_obs2

[1] 0.4

In [144]:
# Hipotesis 3
p_H3_dado_obs2 <- p_obs2_dado_H3 * p_H3 / p_obs2
p_H3_dado_obs2

[1] 0.3

In [145]:
# Hipotesis 4
p_H4_dado_obs2 <- p_obs2_dado_H4 * p_H4 / p_obs2
p_H4_dado_obs2

[1] 0

In [146]:
p_H0_dado_obs2
p_H1_dado_obs2
p_H2_dado_obs2
p_H3_dado_obs2
p_H4_dado_obs2

[1] 0

[1] 0.3

[1] 0.4

[1] 0.3

[1] 0

In [147]:
# Actualizo priors luego de SEGUNDA extracción
p_H0 <- p_H0_dado_obs2 # = 0
p_H1 <- p_H1_dado_obs2
p_H2 <- p_H2_dado_obs2
p_H3 <- p_H3_dado_obs2
p_H4 <- p_H4_dado_obs2

In [148]:
# Saco 1 roja, luego de 1 negra, luego de 1 roja, Hipotesis 0
p_obs3_dado_H0 <- 0
# Hipotesis 1
p_obs3_dado_H1 <- 1/4
# Hipotesis 2
p_obs3_dado_H2 <- 2/4
# Hipotesis 3
p_obs3_dado_H3 <- 3/4
# Hipotesis 4
p_obs3_dado_H4 <- 4/4

p_obs3 <- p_obs3_dado_H0 * p_H0 + p_obs3_dado_H1 * p_H1 + p_obs3_dado_H2 * p_H2 + p_obs3_dado_H3 * p_H3 + p_obs3_dado_H4 * p_H4

In [149]:
# Saco 1 roja, luego de 1 negra, luego de 1 roja, Hipotesis 0
p_H0_dado_obs3 <- p_obs3_dado_H0 * p_H0 / p_obs3
p_H0_dado_obs3

[1] 0

In [150]:
# Hipotesis 1
p_H1_dado_obs3 <- p_obs3_dado_H1 * p_H1 / p_obs3
p_H1_dado_obs3

[1] 0.15

In [151]:
# Hipotesis 2
p_H2_dado_obs3 <- p_obs3_dado_H2 * p_H2 / p_obs3
p_H2_dado_obs3

[1] 0.4

In [152]:
# Hipotesis 3
p_H3_dado_obs3 <- p_obs3_dado_H3 * p_H3 / p_obs3
p_H3_dado_obs3

[1] 0.45

In [153]:
# Hipotesis 4
p_H4_dado_obs3 <- p_obs3_dado_H4 * p_H4 / p_obs3
p_H4_dado_obs3

[1] 0

In [154]:
p_H0_dado_obs3
p_H1_dado_obs3
p_H2_dado_obs3
p_H3_dado_obs3
p_H4_dado_obs3

[1] 0

[1] 0.15

[1] 0.4

[1] 0.45

[1] 0

In [155]:
# Actualizo priors luego de TERCERA extracción
p_H0 <- p_H0_dado_obs3 # = 0
p_H1 <- p_H1_dado_obs3
p_H2 <- p_H2_dado_obs3
p_H3 <- p_H3_dado_obs3
p_H4 <- p_H4_dado_obs3

In [156]:
# Saco 1 roja, luego de 1 roja, luego de 1 negra, luego de 1 roja, Hipotesis 0
p_obs4_dado_H0 <- 0
# Hipotesis 1
p_obs4_dado_H1 <- 1/4
# Hipotesis 2
p_obs4_dado_H2 <- 2/4
# Hipotesis 3
p_obs4_dado_H3 <- 3/4
# Hipotesis 4
p_obs4_dado_H4 <- 4/4

p_obs4 <- p_obs4_dado_H0 * p_H0 + p_obs4_dado_H1 * p_H1 + p_obs4_dado_H2 * p_H2 + p_obs4_dado_H3 * p_H3 + p_obs4_dado_H4 * p_H4

In [157]:
# Saco 1 roja, luego de 1 negra, luego de 1 roja, Hipotesis 0
p_H0_dado_obs4 <- p_obs4_dado_H0 * p_H0 / p_obs4
p_H0_dado_obs4

[1] 0

In [158]:
# Hipotesis 1
p_H1_dado_obs4 <- p_obs4_dado_H1 * p_H1 / p_obs4
p_H1_dado_obs4

[1] 0.06521739

In [159]:
# Hipotesis 2
p_H2_dado_obs4 <- p_obs4_dado_H2 * p_H2 / p_obs4
p_H2_dado_obs4

[1] 0.3478261

In [160]:
# Hipotesis 3
p_H3_dado_obs4 <- p_obs4_dado_H3 * p_H3 / p_obs4
p_H3_dado_obs4

[1] 0.5869565

In [161]:
# Hipotesis 4
p_H4_dado_obs4 <- p_obs4_dado_H4 * p_H4 / p_obs4
p_H4_dado_obs4

[1] 0

In [163]:
p_H0_dado_obs4
p_H1_dado_obs4
p_H2_dado_obs4
p_H3_dado_obs4
p_H4_dado_obs4

[1] 0

[1] 0.06521739

[1] 0.3478261

[1] 0.5869565

[1] 0